src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -253,6 +260,16 @@
     public static $manyMethods = ['belongsToMany', 'morphToMany', 'morphedByMany'];
 
     /**
+     * All of the valid primitive cast types.
+     *
+     * @var array
+     */
+    protected static $primitiveCastTypes = [
```

Comment:
Since Laravel 5.3 is PHP 5.6, you could make this a `CONST`. Do note that HHVM 3.3 doesn't support this though. I'm not sure about later versions.